src/main/java/pw/itr0/kaba/encrypt/PasswordBasedDecrypter.java

```diff
@@ -12,27 +17,61 @@
+
+    /**
+     * Decrypt {@code secret} based on password.
+     *
      * @param secret   secret bytes
+     * @param password decrypting password
      * @return decrypted secret
-     * @throws GeneralSecurityException on decryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                                   or decrypting algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    byte[] decrypt(char[] password, byte[] secret) throws GeneralSecurityException;
+    default byte[] decrypt(byte[] secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return getDecryptionCipher(password).doFinal(secret);
+    }
 
     /**
      * Decrypt {@code secret} based on password.
      * <p>
      * Encrypted secret is encoded as Base64.
      *
-     * @param password decrypting password
      * @param secret   secret bytes (Base64 encoded string)
+     * @param password decrypting password
      * @return decrypted secret
-     * @throws GeneralSecurityException on decryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                                   or decrypting algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
+     */
+    default byte[] decrypt64(String secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return this.decrypt(Base64.getDecoder().decode(secret), password);
+    }
+
+    /**
+     * Wrap {@link InputStream} {@code in} to {@link CipherInputStream} decrypting with {@code password}.
+     *
+     * @param in       {@link InputStream} to wrap in {@link CipherInputStream}
+     * @param password decrypting password
+     * @return decrypting {@link InputStream}
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                             or decrypting algorithm using specified password length is not usable on current JDK
      */
-    default byte[] decrypt64(char[] password, String secret) throws GeneralSecurityException {
-        return this.decrypt(password, Base64.getDecoder().decode(secret));
+    default InputStream wrap(InputStream in, char[] password) throws InvalidKeyException {
+        return new CipherInputStream(in, getDecryptionCipher(password));
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method call to 'getDecryptionCipher' needs "this.".](https://www.codacy.com/app/ryotan/kaba-vault/file/2686265516/issues/source?bid=3298589&fileBranchId=3332567#l75)